core/app/(user)/magic-link/route.ts

```diff
@@ -129,6 +108,14 @@ export async function GET(req: NextRequest) {
 
 	const { user: tokenUser, authTokenType } = tokenSettled.value;
 
+	if (!tokenUser.isVerified) {
```

Comment:
by putting this here in the `magic-link` route, we automatically mark users who go through this route as verified since they would've come from their email. this lets us get the requirement "Forgot password also functions as verification, in case where user signs up, never completes verification, and deletes verification email" for free